### DISCUSSION TITLE
let's talk about react-aria and why it's controversial

### DISCUSSION THREAD

**Discussion by clicktodev** [tailwindlabs/headlessui#3255] [General]
First of all, I would like to thank the team for creating headless ui.
However I believe the decision to go with react-aria breaks the most fundamental interactive element. the `Button`.

I should be able to cancel a click by moving the cursor outside of the button. This is native behavior that is expected by all users.

react-aria button behavior introduces unnecessary incompatibility with standard browser click events.
It feels these button modifications break away from native behavior which introduces problems with accessibility similar to Flutter.

I'd appreciate some feedback on this because It seems from searching on the internet, many people agree with my take.

example:
[Combobox](https://headlessui.com/react/combobox): notice how it registers the click before releasing which doesn't allow the user to cancel a click.

However I don't understand why this behavior isn't consistent across all clickable elements.

**Comment by tmcw** [comment]
Trying out the react-aria [Button element](https://react-spectrum.adobe.com/react-aria/Button.html), I can click the button and drag outside, and that cancels the `onPress` event.

**Comment by laneme** [comment]
I havent tested but a look at [menu-trigger-behavior](https://react-spectrum.adobe.com/react-aria/ComboBox.html#menu-trigger-behavior) in react-aria docs shows me a `menuTrigger` option.

If i set `menuTrigger` to `manual` does that achieve the standard behavior?
